### ISSUE TITLE
Case Fan Detection doesn't work after waking up from sleep or boot

### ISSUE THREAD

**Issue by eduroca** [Rem0o/FanControl.Releases#280]
I mentioned this in another issue some time ago but figured it'd make more sense to put it up as a separate issue.

Right after boot or when waking the computer from boot, Fan Control will show my case fans spinning at 9000 rpm or some other ridiculous value and the speed control will be stuck at 0%. Manually changing the speed will do nothing.

Refreshing sensors will throw a detection error and either remove those fans or keep them while showing no values.

The only thing that works is, with Fan Control open, running HWMonitor. Once HWMonitor loads and starts showing values, Fan Control will start showing normal values for the case fans and I can control them again.

This _doesn't_ happen with the  GPU fan, which always work correctly.

This is on an Asus B450M-A Prime, running Windows 10. FanControl V. 85 (error happens since at least V. 70).

Another use had a a similar issue on an Asus Prime B350-Plus MOBO, so that may be a common link.

Thanks a million for any suggestions!

Edit: originally stated that the CPU fan always worked correctly, turns out it also has the same error.

**Comment by Rem0o** [comment]
Do you have any Asus / manufacturer software running?

**Comment by eduroca** [comment]
> 
> 
> Do you have any Asus / manufacturer software running?

I did a search for "Asus" on all my files and AsusFanControlService.exe turned up. I have no idea when that got installed. It might have been part of AI Suite 3 which I uninstalled ages ago.

As far as I can tell (checked "Details" tab on Task Manager running both as local and admin) it is not a running process nor is it included in the Startup list.

Found a single reference for it in the Registry:
Computer\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Compatibility Assistant\Store

I tried to run it and it just gave me a bunch of missing DLL errors.

I can delete it and see if that helps?

**Comment by Rem0o** [comment]
You should probably delete that,

Also, pushed an update where I refined the wake/sleep routine for fancontrol, see if that helps.

**Comment by eduroca** [comment]
Deleting the files didn't help and the update didn't help but thank you for the followups!

Also, a small amendment to my previous report. I said the CPU fan worked correctly after boot/resume and I was wrong. At least after resuming from sleep it's also showing incorrect speeds.

One more data point: keeping HWMonitor on does not prevent the error on resume from sleep. It seems there's something about the initialization process HWMonitor performs that fixes whatever sensor glitch is messing up FanControl.

**Comment by Rem0o** [comment]
https://github.com/Rem0o/FanControl.Releases/releases/tag/V87

**Comment by eduroca** [comment]
Hey! Unfortunately still no dice, but thanks a million for keeping up with this :)

**Comment by abqnm** [comment]
> Hey! Unfortunately still no dice, but thanks a million for keeping up with this :)

Sorry it's not working for you still. I have been following your issue post since I started having issues with my Asus x570i displaying a huge error page listing all of the fans and sensors as not found after waking from sleep. It seems that one of the last 2 updates (I didn't install v86 since v87 came out hours later and I didn't notice v86 first), something was changed in these last 2 builds which has completely fixed this issue on my Asus board. 

I know the issue wasn't 100% identical but the fixes for your issue helped for me. Hopefully it can be fixed for you too!

**Comment by eduroca** [comment]
> 
> 
> > Hey! Unfortunately still no dice, but thanks a million for keeping up with this :)
> 
> Sorry it's not working for you still. I have been following your issue post since I started having issues with my Asus x570i displaying a huge error page listing all of the fans and sensors as not found after waking from sleep. It seems that one of the last 2 updates (I didn't install v86 since v87 came out hours later and I didn't notice v86 first), something was changed in these last 2 builds which has completely fixed this issue on my Asus board.
> 
> I know the issue wasn't 100% identical but the fixes for your issue helped for me. Hopefully it can be fixed for you too!

Thanks for your data point! It might just be an artifact of my Windows install. If I get a chance at some point I'll try to do a clean install.

Just out of curiosity, would you mind telling me what your BIOS settings are for your chassis fan controls?

**Comment by abqnm** [comment]
> Just out of curiosity, would you mind telling me what your BIOS settings are for your chassis fan controls?

Since it's mini-ITX, it only has one case fan and I have it set in the bios (PWM) to run at 40% to 60C, 65% at 75C then 100% at 80C since it only relies on the CPU sensor but I mainly need it to cool the GPU, whereas with FC it's a combo with the GPU, and that puts it more in line with gaming temps of the 5600x with a tiny cooler. But the curve is more of just a step, since it's only a tiny fan that's not going to cause damage if it isn't working, as the system can still cool itself just with the GPU & CPU fans due to case design. So I just have some "backup" settings set since FC works great at controlling them, but I love being able to control my GPU fans and CPU cooler plus the "case fan" which is a 92mm slim upward facing exhaust. 

Aside from when it briefly wasn't when waking from sleep it's been extremely reliable, but just exiting exiting app by the notification icon (since it was before the change that makes it so errors don't block out the UI, so it was full-screen but there was no obvious way out) and restarting it was enough to fix it every time. 

It's like the app woke up before the controller, and then just immediately gave up. But whatever was tweaked either for your issue or in the last couple builds in general fixed it for me.

Again I'm sorry it isn't fixed for you. Maybe try creating a new windows user account, and "installing" FC there before trying a new install just to see if you have any different behavior, since it wouldn't be linked in any way to your user account (though it's all stored in the same folder as the executable where you extracted it to after download), so you'd be "clean" as if you reinstalled. Almost. 

If that works, I'd say just disable start at boot, delete the folder in your user account and extract the latest version from scratch and run it as if it's new. _Should_ be the same thing. You could also just rename the folder (after disabling start at boot) so the configurations are still there and can be reverted, and then extract and run again as new, which may even be enough. Just some thoughts. I build small form factor systems and service many for a living so I'm sorry if I'm long winded here. I tend to do a lot of troubleshooting, online and off, so when you jumped to windows reinstall, I figured maybe just try the basics first.

**Comment by JordHisu** [comment]
Hey, finally found someone with the same problem! I have an Asus Prime B450M-Gaming/BR, which is basically B450M-A. I'm trying to find the source of the problem, but no clues for now. I thought it might be some conflict with Asus software, so I uninstalled everything, but it didn't solve the problem. I also noticed that the first time I tried to setup custom sensors I was able to select 6 different mobo temperature sensors, and after having this problem, only two show up, or none sometimes. Tried a "fresh install" (aka unzipped and started from zero), to no avail. It's probably not a windows artifact because my system is a week old.

Long story short: GPU RPM measurement and control is always fine, CPU RPM measurement is always locked to 21 for some reason and I can't control, case RPM measurement is a very high boggus and I can't control. When this happens, the program says it has missing sensors or something like that, and trying to use the assisted setup again pairs my controls with "none".

**Comment by empty1958** [comment]
I have a similar issue with a MSI Z690 Pro. The MSI Nvidia is found each time but the MB is not. Tried the reinstall without success. This is annoying but I do like the program overall. It just has the right options. I have relegated myself to shutting it down in task manager and restarting it. So far that has worked every time.

**Comment by Rem0o** [comment]
Added a multi-retry mechanism since, until all sensors are back.
